airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -706,29 +566,10 @@ def run_query(self,
             https://cloud.google.com/bigquery/docs/locations#specifying_your_location
         :type location: str
         """
+        project_id = project_id if project_id is not None else self.project_id
```

Comment:
Again - I think using fallback decorator is nicer as it keeps the project_id logic in one place.